rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -330,7 +330,8 @@ void Recorder::subscribe_topics(
         topic_with_type.first,
         topic_with_type.second,
         serialization_format_,
-        serialized_offered_qos_profiles_for_topic(topic_with_type.first)
+        serialized_offered_qos_profiles_for_topic(topic_with_type.first),
+        ""  // TODO(james-rms): fill in type description hash from endpointInfo
```

Comment:
You could build against the rclcpp PR and fill this out - mark this PR "depends on". Unless you want to merge this sooner and address in a followup PR

Reply:
It's merged :)